src/lc2kFormat.ts

```diff
@@ -1,13 +1,83 @@
 import * as vscode from 'vscode';
+import {TextDecoder} from 'util';
 
 export class Lc2kFormatter implements vscode.DocumentRangeFormattingEditProvider, vscode.DocumentFormattingEditProvider {
   provideDocumentFormattingEdits(document: vscode.TextDocument, options: vscode.FormattingOptions,
                                  token: vscode.CancellationToken): vscode.ProviderResult<vscode.TextEdit[]> {
-    return [];
+    let fullRange = new vscode.Range(0, 0, document.lineCount, 0);
+    return this.provideDocumentRangeFormattingEdits(document, fullRange, options, token);
   }
 
+
   provideDocumentRangeFormattingEdits(document: vscode.TextDocument, range: vscode.Range, options: vscode.FormattingOptions,
                                       token: vscode.CancellationToken): vscode.ProviderResult<vscode.TextEdit[]> {
-    return [];
+    // console.log("Formatting");
+    const config = vscode.workspace.getConfiguration('lc2k');
+    const enable = config.get('format.enable');
+    if (!enable) {
+      return [];
+    }
+    const fillOffset = +config.get('format.filloffset');
+    const jalrOffset = +config.get('format.jalroffset');
+    const noopOffset = +config.get('format.noopoffset');
+    const addNewLine = config.get('format.addNewlineAtEOF');
+    // console.log("jalr offset: " + jalrOffset);
+    // console.log("jalr offset: " + fillOffset);
+
+    var edits = [];
+    var regex;
+    var newText;
+    var offset;
+    for (let i = 0; i < document.lineCount; i++) {
```

Comment:
You need to use the line range given to you by the `range` parameter here. You can get the line number from the `Range`'s beginning position and end position.